### ISSUE TITLE
Stache animation guide and functionality

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/canjs#3864]
__tldr;__ This proposal seeks to create a guide for creating and using animation libraries in `can-stache` and to create the necessary functionality to enable the guide.

### Local animation helper

The guide will show how to create a local animation helper function using jQuery's [fadeIn](http://api.jquery.com/fadein/) and fade out:

```js
import $ from "jquery";
import stache from "can-stache";

// stache.helper labels the function has a helper
// it also makes sure that the `this` is the `helperOptions` with
// this.element and this.event available
module.exports = stache.helper(function(duration, callback){
  $(this.element).fadeIn(duration, callback)
});
```

### Importing animation helpers in the template

Then we will show how to import the animation helper within a stache template:

```
<can-import from="./my-fade-in" value:to="scope.helpers.fadeIn"/>
```

### Importing animation helpers through components 

For folks who are not using steal, lets show how to import helpers for the component:

```js
import fadeIn from "./my-fade-in";

Component.extend({
  view: `...`,
  ViewModel: { ... },
  helpers: {
    fadeIn
  }
})
```

### Global animation helper 

The guide should show how to make a global animation helper too as an aside. This is useful for JSBins that are unable to import with stache.

```js
import $ from "jquery";
import stache from "can-stache";

// stache.helper labels the function has a helper
// it also makes sure that the `this` is the `helperOptions` with
// this.element and this.event available
module.exports = stache.addHelper(function(duration, callback){
  $(this.element).fadeIn(duration, callback)
});
```

### Using a helper 

Once a user has a helper available in stache, we will show them how to use it.

```
<div on:click="fadeIn('slow')">Fade Me In</div>
```

This proposal seeks to finish minimal working "core" set of functionality for `can-stache-animate`.  Critically, this proposal doesn't seek to provide any animations other than a `fadeOut` and `fadeIn` animation which are primarily useful for demonstrating how to build animations.

**Comment by frank-dspeed** [comment]
@justinbmeyer can we simply use callexpressions again? if we do {{fadeIn()}} we should get the element already as far as i remember then in our fadeIn function we can care for the disconnect function to fadeOut.

**Comment by frank-dspeed** [comment]
I copy the Answer from https://github.com/canjs/can-stache/issues/611 =>

@justinbmeyer https://codepen.io/frank-dspeed/pen/KGemLG
example that shows what i mean it is also related to canjs/canjs#4517

it can be also applyed here to render animated sections via callexpressions

**Comment by justinbmeyer** [comment]
Call expressions work. They can be cumbersome
